reporters/index.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+var getReporter = require('jscs/lib/cli-config').getReporter;
+var PluginError = require('gulp-util').PluginError;
+var through = require('through2');
+
+exports.failReporter = require('./fail');
+
+exports.loadReporter = function (reporter) {
+	// we want the function
+	if (typeof reporter === 'function') return reporter;
+
+	// object reporters
+	if (typeof reporter === 'object' && typeof reporter.reporter === 'function') return reporter.reporter;
+
+	// load JSCS built-in or full-path or module reporters
+	if (typeof reporter === 'string' || !reporter) {
+		try {
+			return getReporter(reporter).writer;
+		} catch (err) {}
+	}
+};
+
+exports.reporter = function (reporter, reporterCfg) {
+	reporterCfg = reporterCfg || {};
+
+	if (reporter === 'fail') {
+		return exports.failReporter(reporterCfg);
+	}
+
+	var rpt = exports.loadReporter(reporter);
+
+	if (typeof rpt !== 'function') {
+		throw new PluginError('gulp-jscs', 'Invalid reporter');
+	}
+
+	// return stream that reports stuff
+	return through.obj(function (file, enc, cb) {
+		if (file.jscs && !file.jscs.success) {
+			rpt([file.jscs.errors]);
```

Comment:
This is kind of a hack, because JSCS reporters expect an array containing error collections from all linted files.

Should we add some kind of buffer to pass all error collections at once? Should we buffer only the errors output, or errors and files? What I mean is, holding onto the files until the stream closes would slow the stream down, while holding onto the errors could cause out of order logging. That's why I went with this hacky approach.